### PR TITLE
chore(storage-browser): add MessageControl scaffold

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes
Add MessageControl

**output, with variant 'error'**

```html
<div class="storage-browser__message" role="alert">
  <svg
    aria-hidden="false"
    width="24"
    height="24"
    viewBox="0 -960 960 960"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    class="storage-browser__message__icon"
    aria-label="Error"
  >
    <path
      d="M480-280q17 0 28.5-11.5T520-320q0-17-11.5-28.5T480-360q-17 0-28.5 11.5T440-320q0 17 11.5 28.5T480-280Zm-40-160h80v-240h-80v240Zm40 360q-83 0-156-31.5T197-197q-54-54-85.5-127T80-480q0-83 31.5-156T197-763q54-54 127-85.5T480-880q83 0 156 31.5T763-763q54 54 85.5 127T880-480q0 83-31.5 156T763-197q-54 54-127 85.5T480-80Zm0-80q134 0 227-93t93-227q0-134-93-227t-227-93q-134 0-227 93t-93 227q0 134 93 227t227 93Zm0-320Z"
      fill="currentColor"
    ></path>
  </svg>
  <div class="storage-browser__message__content">
    Lorem Ipsum is simply dummy text of the printing and typesetting industry.
  </div>
  <button
    class="storage-browser__message__dismiss"
    aria-label="Dismiss message"
  >
    <svg
      aria-hidden="true"
      width="24"
      height="24"
      viewBox="0 -960 960 960"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      class="storage-browser__message__dismiss__icon"
    >
      <path
        d="m256-200-56-56 224-224-224-224 56-56 224 224 224-224 56 56-224 224 224 224-56 56-224-224-224 224Z"
        fill="currentColor"
      ></path>
    </svg>
  </button>
</div>
```

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx [hunk 1]
```diff
@@ -1,18 +1,115 @@
...
+    ) : null;
+  }
+);
+
+const MessageContent = withBaseElementProps(
+  View,
+  ({ className = `${BLOCK_NAME}__content`, ...props }) => ({
+    ...props,
+    className,
+  })
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx:84]: Just noting an object can be passed as `defaultProps` if no logic is required in the callback

File: packages/react-storage/src/components/StorageBrowser/Views/Controls/__tests__/Message.spec.tsx [hunk 2]
```diff
@@ -1,9 +1,43 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 import { MessageControl } from '../Message';
 
 describe('MessageControl', () => {
   it('renders a `MessageControl`', () => {
-    expect(render(<MessageControl />).container).toBeDefined();
+    render(<MessageControl />);
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/Views/Controls/__tests__/Message.spec.tsx:7]: 🙏

File: packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx [hunk 3]
```diff
@@ -1,18 +1,115 @@
...
+> extends Pick<T, 'Icon'> {
+  ({ variant }: MessageControlProps): React.JSX.Element;
+  Container: T['View'];
+  Content: T['View'];
+  Dismiss: MessageDismissControl<T>;
 }
 
-export const MessageControl: MessageControl = () => {
-  return <>Message</>;
+export const MessageControl: MessageControl = ({ variant }) => {
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx:98]: Just a heads up: `messageType` will be provided by the control context in the future, no need to change this though

File: packages/react-storage/src/components/StorageBrowser/context/elements/definitions.ts [hunk 4]
```diff
@@ -25,7 +25,7 @@ export interface StorageBrowserElements {
 }
 
 export type PaginateVariant = `paginate-${'next' | 'current' | 'previous'}`;
-
+export type MessageVariant = 'info' | 'success' | 'error' | 'warning';
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/context/elements/definitions.ts:28]: What is your opinion on scoping control specific `Variant` types this way? Added `PaginateVariant` to narrow the allowed `variant` values in `PaginateControl` but not sure it makes sense

**Review by calebpollman — APPROVED** [review]
Thanks for adding the unit tests! Left some non-blocking _comments_ (opinions?) 🚢

**Review by timngyn — APPROVED** [review]

File: packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx [hunk 5]
```diff
@@ -1,18 +1,115 @@
...
+        break;
+    }
+
+    return variant ? (
+      <IconElement
+        className={`${BLOCK_NAME}__icon`}
+        aria-hidden="false"
+        aria-label={ariaLabel}
+        variant={variant}
+        {...props}
```
[the remark this answers is not in the document]
Comment (hbuchel) [reply packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx:68]: Ah that makes sense now, thank you! Updated in the latest.

**Review by timngyn — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**